psydac/core/bsplines.py

```diff
@@ -273,6 +281,8 @@ def basis_funs_all_ders( knots, degree, x, span, n ):
         ders[k,:] = ders[k,:] * r
         r = r * (degree-k)
 
+    if normalization == 'M':
+        ders *= scaling
```

Comment:
You can avoid storing the `scaling` array (and also spare some divisions) with the following code:
```suggestion
    if normalization == 'M':
        ders *= [(degree + 1) / (knots[i + degree + 1] - knots[i]) \
                 for i in range(span - degree, span + 1)]
```

Reply:
Fixed in [3d32ea6](https://github.com/pyccel/psydac/pull/193/commits/3d32ea6ea706b76e80e6d0f74ca8acf36c47be3b)